File: app-java/src/main/java/com/example/app_java/Models/Card.java
package com.example.app_java.Models;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Card {
    // Represents a card in the memory game.

    @JsonProperty("type")
    private String type; // Type of the card, i.e., image, color, number, etc.

    @JsonProperty("flipped")
    private Boolean flipped; // Indicates whether the card is hidden or flipped.

    // Constructor
    public Card() {
    }

    // Getter and setter for type
    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    // Getter and setter for flipped
    public Boolean getFlipped() {
        return flipped;
    }

    public void setFlipped(Boolean flipped) {
        this.flipped = flipped;
    }
}

File: complete-app-nosql/app/Models/Card.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Text.Json.Serialization;

namespace app.Models
{
    // Represents a card in the memory game.
    public class Card
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public required string Id { get; set; }

        // Gets or sets the type of the card, i.e. image, color, number, etc.
        [JsonPropertyName("type")]
        public required string Type { get; set; }

        // Gets or sets a value indicating whether the card is hidden or flipped.
        [JsonPropertyName("flipped")]
        public required bool Flipped { get; set; }

        public ICollection<GameCard> GameCards { get; set; } = new List<GameCard>();
    }
}